src/main/java/core/basesyntax/strategy/impl/SupplyOperation.java

```
@@ -0,0 +1,17 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+
+public class SupplyOperation implements OperationHandler {
+    @Override
+    public void handle(FruitTransaction transaction) {
+        if (transaction == null) {
+            throw new RuntimeException("Transaction cannot be null.");
+        }
+        int currentQuantity = Storage.getFruitQuantity(transaction.getFruit());
```

You validate `transaction` for null which is good, but you do not validate `transaction.getFruit()` and `transaction.getQuantity()`. The checklist requires defensive handling of invalid input (null parameters / incorrect data). Add checks: `if (fruit == null || fruit.trim().isEmpty()) throw new RuntimeException("Fruit name cannot be null or empty.");` and `if (quantity < 0) throw new RuntimeException("Quantity cannot be negative.");` before using Storage. This ensures handlers fail fast with informative errors.